File: python/scripts/create_user.py
import os
from travelperk_http_python.builder.builder import build
from utils import write_output
travelperk = build(os.environ["SANDBOX_API_KEY"], True)
user = travelperk.scim().users().make(
    'test@api.probe',
    True,
    'Test',
    'Api Probe'
).set_locale("en").set_title("test_user").save()
write_output("create_user", [
    user.schemas[0],
    user.schemas[1],
    user.schemas[2],
    user.name.given_name,
    user.name.family_name,
    user.name.middle_name,
    user.name.honorific_prefix,
    user.locale,
    user.preferred_language,
    user.title,
    user.external_id,
    user.id,
    user.groups,
    user.active,
    user.user_name,
    len(user.phone_numbers),
    user.enterprise_extension.cost_center,
    user.travelperk_extension.gender,
    user.travelperk_extension.date_of_birth,
    user.travelperk_extension.travel_policy,
    user.travelperk_extension.invoice_profiles[0].value,
    user.travelperk_extension.emergency_contact.name,
    user.travelperk_extension.emergency_contact.phone,
    len(user.travelperk_extension.approvers),
])

File: php/scripts/create_user.php
<?php
require_once(__DIR__."/../vendor/autoload.php");
require_once("utils.php");
use Namelivia\TravelPerk\ServiceProvider;
$travelperk = (new ServiceProvider())->build(getenv("SANDBOX_API_KEY"), true);
$user = $travelperk->scim()->users()->make(
	'test@api.probe',
	true,
	'Test',
	'Api Probe'
)->setLocale('en')->setTitle('test_user')->save();
write_output("create_user", [
	$user->schemas[0],
	$user->schemas[1],
	$user->schemas[2],
	$user->name->givenName,
	$user->name->familyName,
	$user->name->middleName,
	$user->name->honorificPrefix,
	$user->locale,
	$user->preferredLanguage,
	$user->title,
	$user->externalId,
	$user->id,
	$user->groups,
	$user->active,
	$user->userName,
	count($user->phoneNumbers),
	$user->enterpriseExtension->costCenter,
	$user->travelperkExtension->gender,
	$user->travelperkExtension->dateOfBirth,
	$user->travelperkExtension->travelPolicy,
	$user->travelperkExtension->invoiceProfiles[0]->value,
	$user->travelperkExtension->emergencyContact->name,
	$user->travelperkExtension->emergencyContact->phone,
	count($user->travelperkExtension->approvers),
]);
